modules/150-user-authn/webhooks/validating/user

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Flant JSC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source /shell_lib.sh
+
+function __config__(){
+  cat <<EOF
+configVersion: v1
+kubernetes:
+  - name: users
+    apiVersion: deckhouse.io/v1
+    kind: User
+    queue: "users"
+    group: main
+    executeHookOnEvent: []
+    executeHookOnSynchronization: false
+    keepFullObjectsInMemory: false
+    jqFilter: |
+      {
+        "name": .metadata.name,
+        "email": .spec.email,
+        "userID": .spec.userID
+      }
+kubernetesValidating:
+- name: users-unique.deckhouse.io
+  group: main
+  rules:
+  - apiGroups:   ["deckhouse.io"]
+    apiVersions: ["*"]
+    operations:  ["CREATE", "UPDATE"]
+    resources:   ["users"]
+    scope:       "Cluster"
+EOF
+}
+
+function __main__() {
+  userName=$(context::jq -r '.review.request.object.metadata.name')
+  email=$(context::jq -r '.review.request.object.spec.email')
+  userID=$(context::jq -r '.review.request.object.spec.userID')
+
+  if userWithTheSameEmail="$(context::jq -er --arg name "$userName" --arg email "$email" --arg userid "$userID" '.snapshots.users[].filterResult | select(.name != $name) | select(.email == $email) | .name' 2>&1)"; then
+    cat <<EOF > "$VALIDATING_RESPONSE_PATH"
+{"allowed":false, "message":"Email must be unique. User '$userWithTheSameEmail' is already using email '$email'" }
```

Comment:
It seems like there should be different logic for updates because it will always find the updated email in the snapshot if a user changes other fields.